### PR TITLE
perf: do FAT LTO and singular `codegen-unit` on release

### LINKED ISSUE #224: PGO + Bolt
parseable right now does not support building with more advanced optimization techniques like [PGO](https://en.wikipedia.org/wiki/Profile-guided_optimization) and [BOLT](https://github.com/llvm/llvm-project/tree/main/bolt). This tooling has an increasing adoption in the community as a tool to additionally optimize programs. With this tooling, there is a huge chance to gain even more performance "for free".

Here I suggest considering an option at least to play with LTO ("fat" version) + PGO + Bolt pipeline (or any combination of them) and test, does it give a performance to the project or not. If yes, would be awesome to have prebuilt binaries with more advanced optimization from the scratch. Also, for the users will be helpful to have the ability to tweak manually their own binaries to their own workloads with the integrated into the build scripts functionality.

Since the project is quite small I do not expect (at least yet) significant improvement from the PGO right now. But at least a good option to try :)

Also, there are some caveats to consider like:
- Significantly increased build times
- BOLT could be still unstable (or even broken) on some architectures

Links:
- Rust experience with LTO + PGO + BOLT: https://kobzol.github.io/rust/rustc/2022/10/27/speeding-rustc-without-changing-its-code.html

### PR DESCRIPTION
**Pull request by de-sh**
Fixes #224 .

### Description

Optimize release builds by building them in a special profile `release-lto` that:
1. Enables `FAT LTO`
2. Uses `codegen-unit = 1`
This is intended to improve the quality of parseable binary compiled at release, in terms of performance and size.

<hr>

This PR has:
- [ ] been tested to ensure log ingestion and log query works.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

### CODE REVIEW

**Comment by de-sh** [comment]
I have read the CLA Document and I hereby sign the CLA

File: Cargo.toml [hunk 1]
```diff
@@ -8,3 +8,7 @@ resolver = "2"
 # - https://github.com/apache/arrow-rs/pull/6260
 # But a new version hasn't been published to crates.io for this yet. So, we are using this patch temporarily.
 object_store = { git = "https://github.com/apache/arrow-rs.git", rev = "23b6ff9f432e8e29c08d47a315ba0b7cb8758225" }
+
+[profile.release]
+lto = "thin"
+codegen-units = 1
```
Comment (nitisht) [inline Cargo.toml:14]: I think it is worth trying with FAT as well and see if we get any perf advantages. cc @nikhilsinhaparseable

File: Cargo.toml [hunk 2]
```diff
@@ -8,3 +8,7 @@ resolver = "2"
 # - https://github.com/apache/arrow-rs/pull/6260
 # But a new version hasn't been published to crates.io for this yet. So, we are using this patch temporarily.
 object_store = { git = "https://github.com/apache/arrow-rs.git", rev = "23b6ff9f432e8e29c08d47a315ba0b7cb8758225" }
+
+[profile.release]
+lto = "thin"
```
Comment (de-sh) [inline Cargo.toml:13]:
```suggestion
```

Just realized that `"thin"` is default 🃏

**Comment by de-sh** [comment]
Decision on this front, we will move these changes to a specific profile `release-lto` that will be used during compilation for binary generation on GitHub release.

**Review by nitisht — APPROVED** [review]

**Pull request merged**